File: app/src/main/java/co/fddittmar/planbuilder/view/adapter/ExercisesAdapter.java
package co.fddittmar.planbuilder.view.adapter;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

import co.fddittmar.planbuilder.R;
import co.fddittmar.planbuilder.data.model.Exercise;

/**
 * Adapter to represent a list of Exercises items
 */

public class ExercisesAdapter extends RecyclerView.Adapter<ExercisesAdapter.ViewHolder> {
    private List<Exercise> exercises;


    public ExercisesAdapter(List<Exercise> exercises ){
        this.exercises = exercises;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater
                .from( parent.getContext() )
                .inflate(R.layout.item_exercise, parent, false);
        return new ViewHolder( view );
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        holder.setData( exercises.get( position ) );
    }

    @Override
    public int getItemCount() {
        return exercises.size();
    }


    class ViewHolder extends RecyclerView.ViewHolder {
        private TextView tvName;
        private TextView tvReps;
        private TextView tvWeight;

        private ViewHolder(View itemView) {
            super(itemView);

            tvName = (TextView) itemView.findViewById(R.id.tv_exercise_name);
            tvReps = (TextView) itemView.findViewById(R.id.tv_exercise_reps);
            tvWeight = (TextView) itemView.findViewById(R.id.tv_exercise_weight);
        }

        private void setData( Exercise exercise ){

            tvName.setText( exercise.getName() );
            tvReps.setText( String.valueOf(exercise.getReps()) );
            tvWeight.setText( String.valueOf(exercise.getWeight()) );
        }
    }
}

File: app/src/main/java/co/fddittmar/planbuilder/view/adapter/ExercisesAdapter.kt
package co.fddittmar.planbuilder.view.adapter

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import co.fddittmar.planbuilder.R
import co.fddittmar.planbuilder.data.model.Exercise

/**
 * Adapter to represent a list of Exercises items
 */

class ExercisesAdapter(private val exercises: List<Exercise>) : RecyclerView.Adapter<ExercisesAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater
                .from(parent.context)
                .inflate(R.layout.item_exercise, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.setData(exercises[position])
    }

    override fun getItemCount(): Int {
        return exercises.size
    }


    inner class ViewHolder constructor(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val tvName: TextView = itemView.findViewById(R.id.tv_exercise_name) as TextView
        private val tvReps: TextView = itemView.findViewById(R.id.tv_exercise_reps) as TextView
        private val tvWeight: TextView = itemView.findViewById(R.id.tv_exercise_weight) as TextView

        fun setData(exercise: Exercise) {

            tvName.text = exercise.name
            tvReps.text = exercise.reps.toString()
            tvWeight.text = exercise.weight.toString()
        }
    }
}
